src/app/components/TransactionDialog/index.tsx

```diff
@@ -0,0 +1,256 @@
+        <>
+          <h1>{getTransactionTitle(tx.status, action)}</h1>
+          <WalletLogo wallet={wallet} />
+          <img
+            src={txFailed}
+            alt="failed"
+            className="tw-w-8 tw-mx-auto tw-mb-4 tw-opacity-75"
+          />
+          <p className="tw-text-center tw-text-warning">
+            <Trans i18nKey={translations.transactionDialog.txStatus.aborted} />
+          </p>
+          {wallet === 'ledger' && (
+            <p className="tw-text-center tw-text-warning">
+              {t(translations.transactionDialog.txStatus.abortedLedger)}
+            </p>
+          )}
+          {tx.retry && (
+            <ActionButton
+              text={t(translations.transactionDialog.pendingUser.retry)}
+              onClick={tx.retry}
+              className={cn(
+                styles.submit,
+                'tw-flex tw-items-center tw-justify-center tw-h-12 tw-rounded-lg tw-w-80 tw-mx-auto',
+              )}
+              textClassName="tw-inline-block tw-text-lg"
+            />
+          )}
+        </>
+      )}
+      {[TxStatus.PENDING, TxStatus.CONFIRMED].includes(tx.status) && (
+        <>
+          <h1>{getTransactionTitle(tx.status, action)}</h1>
+
+          <div className={styles.styledStatus}>
+            <img
+              src={getStatusImage(tx.status)}
+              className={cn({
+                'tw-animate-spin': tx.status === TxStatus.PENDING,
+              })}
+              alt="Status"
+            />
+          </div>
+          <div className="tw-px-8">
+            {finalMessage && (
+              <div className="tw-w-full tw-rounded-xl tw-bg-gray-2 tw-px-6 tw-py-4 tw-mb-4">
+                {finalMessage}
+              </div>
+            )}
+
+            {tx.txHash && (
+              <div className="tw-w-full tw-rounded-xl tw-bg-gray-2 tw-grid tw-grid-cols-2 tw-px-6 tw-py-4">
+                {fee && (
+                  <>
+                    <span className="tw-text-sm">
+                      {t(
+                        translations.perpetualPage.tradeForm.labels.tradingFee,
+                      )}
+                    </span>
+                    <span>{fee}</span>
+                  </>
+                )}
+                <span className="tw-text-sm">Tx ID:</span>
+                <LinkToExplorer
+                  txHash={tx.txHash}
+                  text={prettyTx(tx.txHash)}
+                  className="tw-text-blue tw-underline"
+                />
+              </div>
+            )}
+          </div>
+
+          <ActionButton
+            onClick={close}
+            text={t(translations.common.close)}
+            className={cn(
+              styles.submit,
+              'tw-flex tw-items-center tw-justify-center tw-h-12 tw-rounded-lg tw-w-80 tw-mx-auto tw-mt-16',
+            )}
+            textClassName="tw-inline-block tw-text-lg"
+          />
+        </>
+      )}
+    </Dialog>
+  );
+};
+
+const getWalletName = wallet => {
```

Comment:
All of those helper methods could be extract to a separate file (e.g. `utils.ts`) because I think this component is huge for no reason.